[CALCITE-826] Use ProjectFactory in AggregateJoinTranposeRule and FilterJoinRule

jcamachor opened the pull request.

julianhyde commented:
Currently test suite gives lots of StackOverflowErrors. See jira case for details.

jcamachor commented:
@julianhyde , new pull request addresses the issues; we were getting into an infinite loop due to the calls because the default project factory was calling again RelOptUtil to create the Project operator.
